wallet/src/de/schildbach/wallet/ui/payments/ReceiveFragment.kt

```diff
@@ -64,7 +65,8 @@ class ReceiveFragment : Fragment(R.layout.fragment_receive) {
                 }
             )
         }
-
+        enterAmountViewModel.setMaxAmount(Coin.ZERO)
+        enterAmountViewModel.setMinAmount(Coin.ZERO)
```

Comment:
If the user goes to the Gift Card purchase flow, these limits will be set to match min/max of the card and the wallet balance and will limit what can be entered in the Receive screen.  Therefore we need to reset the limits to zero which allows any number to be entered.

Not sure if there is a different way to clear the values with enterAmountViewModel by attaching it to a particular navigation graph.  It needs to have a scope larger than a single fragment.  With activityViewModels it stays alive while MainActivity is alive.

Reply:
I remember thinking on this problem in the context of leaking EnterAmountFragment. We might need a separate story if we want to fix this at some point.

Moving EnterAmountFragment to Jetpack Compose can also solve the problem, since we can encapsulate both UI and behavior into a single component that's built from smaller ones. In this case, Gift Card Buy screen doesn't actually need to be shared with other similar screens like Send, Coinbase and Crowdnode since it's quite different (fiat-only input, fixed denominations), and instead it can be built anew but with same sub-components.